util/addrmapgen.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import pathlib
+import re
+import csv
+import operator
+
+from pytablewriter import MarkdownTableWriter
+from pytablewriter.style import Style
+from mako.template import Template
+
+
+# Compile a regex to trim trailing whitespaces on lines.
+re_trailws = re.compile(r'[ \t\r]+$', re.MULTILINE)
+
+
+def write_template(tpl_path, outdir, **kwargs):
+    if tpl_path:
+        tpl_path = pathlib.Path(tpl_path).absolute()
+        print(tpl_path)
+        if tpl_path.exists():
+            tpl = Template(filename=str(tpl_path))
+            with open(outdir / tpl_path.with_suffix("").name, "w") as file:
+                code = tpl.render_unicode(**kwargs)
+                code = re_trailws.sub("", code)
+                file.write(code)
+        else:
+            # print(outdir, tpl_path)
+            raise FileNotFoundError
+
+
+def get_size_string(size):
+    size_str = ""
+    if size >= 1024*1024*1024*1024:
+        scaled_size = size/(1024*1024*1024*1024)
+        size_str = "{:.1f} TB".format(scaled_size)
+    elif size >= 1024*1024*1024:
+        scaled_size = size/(1024*1024*1024)
+        size_str = "{:.1f} GB".format(scaled_size)
+    elif size >= 1024*1024:
+        scaled_size = size/(1024*1024)
+        size_str = "{:.1f} MB".format(scaled_size)
+    elif size >= 1024:
+        scaled_size = size/(1024)
+        size_str = "{:.1f} KB".format(scaled_size)
+    else:
+        scaled_size = size
+        size_str = "{:.1f} B".format(scaled_size)
+    return size_str
+
+
+def get_label_pos(max_nr_merged_entries):
+    half = max_nr_merged_entries / 2
+    (idx0, idx1) = (0, 0)
+    if (max_nr_merged_entries % 2) == 0:
+        idx0 = int(half-1)
+        idx1 = int(half)
+    else:
+        idx0 = int(half)
+        idx1 = int(half)+1
+    return (idx0, idx1)
+
+
+def main():
```

Comment:
Bit of a nitpick, but I prefer to parse the args in the `if __name__ == '__main__'` block and pass them to the main function unpacked; this gives `main()` (or whatever invoked function) a clear interface with (if type annotations are used) well-understood args.

Reply:
Thanks for the input! I have no strong feelings on this, so I can implement your proposed changes.